ckanext/data_qld/user_creation/validators.py

```diff
@@ -6,8 +6,10 @@
 def data_qld_user_name_validator(key, data, errors, context):
     user = toolkit.c.userobj
     is_sysadmin = user is not None and user.sysadmin
-
-    if not is_sysadmin and 'publisher' in data[key].lower():
+    # If the username is existing and haven't changed or its from reset password form, skip validation
+    if user is None or user.name.lower() == data[key].lower():
```

Comment:
This could be simpler, just adding the equality check as a third condition on the existing `if`.

Reply:
if the validation is called from reset form, we dont have value for `user` so thats why i needed the extra if statement

Reply:
But won't that also mean that the validation doesn't work when registering a new user?

Reply:
yes, you are right, need to think another option

Reply:
When performing a password reset, the `context` object should contain a `reset_password` flag. See https://github.com/qld-gov-au/ckan/blob/f2524f9029e62cfb8014159d261e5b62625bd9dd/ckan/views/user.py#L634